src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/Transformer.kt

```diff
@@ -148,8 +148,8 @@ fun CourtEventChargeResponse.toDpsMigrationCharge(
   linkedSourceCase: CourtCaseResponse?,
   isSourceCase: Boolean,
 ): MigrationCreateCharge {
-  val linkedCourtEvent =
-    linkedSourceCase?.courtEvents?.firstOrNull { it.courtEventCharges.firstOrNull { it.offenderCharge.id == chargeId } != null }
+  val lastSourceCourtEventForCharge =
+    linkedSourceCase?.courtEvents?.lastOrNull { appearance -> appearance.courtEventCharges.any { charges -> charges.offenderCharge.id == chargeId } }
   return MigrationCreateCharge(
```

Comment:
In NOMIS the source court event has no meaning; until DPS changes requirements at least make the more sensible with the last event rather than first (though neither are correct IMHO)